radar-auth-jersey/src/main/kotlin/org/radarbase/auth/jersey/EcdsaJwtTokenValidator.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2019. The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * See the file LICENSE in the root of this repository.
+ */
+
+package org.radarbase.auth.jersey
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.JWTVerifier
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.AlgorithmMismatchException
+import com.auth0.jwt.exceptions.JWTVerificationException
+import com.auth0.jwt.exceptions.SignatureVerificationException
+import org.radarcns.auth.exception.ConfigurationException
+import org.radarbase.auth.jersey.impl.JwtAuth
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.security.KeyFactory
+import java.security.KeyStore
+import java.security.PublicKey
+import java.security.interfaces.ECPublicKey
+import java.security.interfaces.RSAPublicKey
+import java.security.spec.X509EncodedKeySpec
+import java.util.*
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.core.Context
+
+class EcdsaJwtTokenValidator constructor(@Context private val config: AuthConfig) : AuthValidator {
```

Comment:
Now I am a little confused. Is this meant to replace the `radar-auth`? I see a lot of redundancy between `radar-auth` and this. My previous comment was based on the assumption that `radar-auth-jersey` would be a sub-library of `radar-auth`.

Reply:
This validator is only used if a raw ECDSA provider like Keycloak is used.

Reply:
I have explicitly set Auth to extend JwtRadarToken now. This way, the relation between the authentication in radar-auth and radar-auth-jersey is more explicit.